packages/core/src/components/Tabs/Tabs.test.jsx

```diff
@@ -109,17 +109,48 @@ describe('Tabs', function() {
 
     it('calls onChange', () => {
       const onChangeMock = jest.fn();
-      const data = shallowRender(
+      const data = render(
         {
           onChange: onChangeMock,
           selectedId: 'panel-1'
         },
-        children
+        children,
+        true
       );
 
       data.wrapper.setState({ selectedId: 'panel-2' });
 
       expect(onChangeMock.mock.calls.length).toBe(1);
     });
+
+    it('selects the second panel on right arrow keyDown', () => {
+      const data = render(
+        { defaultSelectedId: 'panel-1' },
+        children,
+        true
+      );
+      const panels = data.wrapper.find('TabPanel');
+      const tabs = data.wrapper.find('Tab');
+
+      tabs.first().simulate('keyDown', { key: 'ArrowRight' });
+
+      expect(panels.at(1).prop('selected')).toBe(true);
+      expect(tabs.at(1).prop('selected')).toBe(true);
+    });
```

Comment:
@1Copenut I think I finally figured out why these new tests are failing. Looks like it's related to this: http://airbnb.io/enzyme/docs/guides/migration-from-2-to-3.html#for-mount-updates-are-sometimes-required-when-they-werent-before

TLDR: 

1. On line 132, we get the `panels` in their initial state (Panel 1 is selected)
2. On line 135, we change the state in the component. `panels` doesn't auto-update though, so it still thinks Panel 1 is selected, when in reality Panel 2 is now selected.

The solution appears to be doing the `data.wrapper.find...` again after the state is changed so that we get the components in their updated state.